deps/rabbit/src/rabbit_quorum_queue.erl

```diff
@@ -488,14 +519,73 @@ handle_tick(QName,
                                                        rabbit_fifo:make_purge_nodes(Stale)),
 
                               ok
-                      end
+                      end,
+
+                      %% ----------
+                      maybe_grow_qq_members(QName, LastActive)
                   catch
                       _:_ ->
                           ok
                   end
           end),
     ok.
 
+maybe_grow_qq_members(QName, LastActive) ->
+    {ok, Q} = rabbit_amqqueue:lookup(QName),
+    % Rely on LastActive and compare it with system_time, to figure out intervall
+    case check_if_grow(LastActive) of
+        false ->
+            rabbit_log:info(">>>> TICK do nothing",[]),
+            ok;
+        true ->
+            rabbit_log:info(">>>> TICK check members",[]),
+            Running = rabbit_nodes:all_running(),
+            {ok, Members, _} = ra:members(amqqueue:get_pid(Q)),
+            MemberNodes = [Node || {_, Node} <- Members],
+            New = Running -- MemberNodes,
+            Size = case rabbit_policy:get(<<"target-group-size">>, Q) of
+                       undefined-> 0;
+                       V -> V
+                   end,
+            CurrentSize = length(MemberNodes),
+            case {CurrentSize < Size, New} of
+                {_, []} ->
+                    %% No new nodes to grow towards, regardless if QQ is below target
+                    ok;
+                {true, NewNodes} ->
+                    %% Sort/filter nodes before sending them to NewNodes, i.e perhaps
+                    %% Take node load or availability zones into account?
+                    add_members(Q, grow_order_sort(NewNodes), 5000, Size - CurrentSize);
+                {_,_} ->
+                    %%Target size already achieved
+                    ok
+            end
+    end.
+
+%% Sort nodes based on number of QQ they have
+grow_order_sort(Nodes) ->
+    QueueLenFun =
+        fun(Node) ->
+                length([Q || Q <- rabbit_amqqueue:list_by_type(quorum),
+                             amqqueue:get_state(Q) =/= crashed,
+                             lists:member(Node, rabbit_amqqueue:get_quorum_nodes(Q))])
+        end,
+    lists:keysort(
+          2,
+          [{Node, QueueLenFun(Node)} || Node <- Nodes]).
+
+%% Return true every aprox GROW_CHECK_TIMEOUT sec
+check_if_grow(LastActive) ->
```

Comment:
@michaelklishin @kjnilsson 
OK, so my idea here is to instead use 'some' timestamp as a reference, and since a queue has a last_active timestamp, I use that. With it I can calculate an approximate interval check. Approximate as in, if you have say a TICK_TIMEOUT of 4 seconds, and you set GROW_CHECK_TIMEOUT to 30 second, the check will be every 32 second, as I round 'up' to nearest multiple of TICK_TIMEOUT to GROW_CHECK_TIMEOUT

Reply:
The last_active time stamp does get reset when something happens for the queue, such as giving it a policy. So adding a policy to existing queues might trigger the interval check a bit early. But for adding nodes etc it should be true to the interval config